packages/apm/src/integrations/tracing.ts

```diff
@@ -505,6 +572,16 @@ export class Tracing implements Integration {
 
     Tracing._log('[Tracing] Adding & adjusting spans using Performance API');
 
+    // FIXME: depending on the 'op' directly is brittle.
+    if (transactionSpan.op === 'pageload') {
+      // Force any pending records to be dispatched.
+      forceLCP();
+      if (lcp) {
+        // Set the last observed LCP score.
+        transactionSpan.setData('_sentry_extra_metrics', JSON.stringify({ lcp }));
+      }
+    }
```

Comment:
Note that the reference implementation sends the LCP when the page visibility changes to hidden. That happens when a user switches tabs, navigates away, etc.

Instead here, as per Daniel's suggestion, we send the current known LCP at the time of sending the "idle transaction".
Local exploratory tests suggest that it is easy to miss the proper LCP entry (either get nothing or get an earlier element).

We may discuss this behavior and decide whether we're happy with it for starters.

Two alternative ways forward are delaying the submission of the pageload transaction (wait for page visibility hidden event) or sending a separate transaction for LCP (in which case we would benefit from a being able to send a trimmed down payload, e.g. without breadcrumbs and other non-essential or redundant data).

---

<details>
<summary>Snippet from the reference implementation</summary>

![image](https://user-images.githubusercontent.com/88819/83315385-e81b2380-a21f-11ea-92e5-ebab209800b9.png)

</details>